game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -109,21 +109,23 @@ public static void main(final String[] args) {
         "UI client launcher invoked from headless environment. This is current prohibited by design to "
             + "avoid UI rendering errors in the headless environment.");
 
-    ErrorHandler.registerExceptionHandler();
     ClientSetting.initialize();
 
     if (!ClientSetting.USE_EXPERIMENTAL_JAVAFX_UI.booleanValue()) {
-      final Console console = new Console();
-      final SimpleFormatter formatter = new SimpleFormatter();
-      LogManager.getLogManager().getLogger("").addHandler(new ConsoleHandler(
-          logMsg -> {
-            if (logMsg.getLevel().intValue() > Level.INFO.intValue()) {
-              ErrorMessage.show(logMsg.getMessage());
-            }
-            console.append(formatter.format(logMsg));
-          }));
-      ErrorMessage.enable();
-      Interruptibles.await(() -> SwingAction.invokeAndWait(LookAndFeel::setupLookAndFeel));
+      Interruptibles.await(() -> SwingAction.invokeAndWait(() -> {
+        LookAndFeel.setupLookAndFeel();
+        final Console console = new Console();
+        final SimpleFormatter formatter = new SimpleFormatter();
+        LogManager.getLogManager().getLogger("").addHandler(new ConsoleHandler(
+            logMsg -> {
+              if (logMsg.getLevel().intValue() > Level.INFO.intValue()) {
+                ErrorMessage.show(logMsg.getMessage());
+              }
+              console.append(formatter.format(logMsg));
+            }));
+        ErrorHandler.registerExceptionHandler();
```

Comment:
Hmm, my rule of thumb is to minimize what is run on EDT thread to what is really needed there. Both for performance and clarity. The ErrorHandler.registerExceptionhandler does not need to  on EDT thread. Also, it seems like this registration should remain early in the startup process in case we have odd environmental errors

Reply:
@ssoloff I understand this swing related, but did not catch why it must be run on EDT now.

Reply:
> I understand this swing related, but did not catch why it must be run on EDT now.

As mentioned above, this method was running on the EDT prior to #3715.

I agree that it's probably not _required_ to run this method on the EDT.  But, as I said in a previous comment, this PR restores the ordering of calls as they were prior to #3715.  I don't know if moving the call to this method before or after its current position is going to change anything, and I don't really have the inclination to find out.  However, I do know that this order works because, prior to #3715, it ran without issue since the beginning of the year.

Reply:
de18891de

Reply:
thanks